prov/shm/src/smr_ep.c

```diff
@@ -1421,6 +1426,8 @@ static int smr_ep_ctrl(struct fid *fid, int command, void *arg)
 			srx->owner_ops = smr_get_peer_srx(ep)->owner_ops;
 			srx->ep_fid.fid.context =
 				smr_get_peer_srx(ep)->ep_fid.fid.context;
+			/* ep->srx was allocated earlier during ep bind */
+			free(ep->srx);
```

Comment:
Instead of freeing it here, can we just remove the realloc here (line 1424)? We also already set the peer ops, owner ops, context, and fid, so we should be able to remove the entire section (5 lines) above your addition as well as the one line below it. The only thing we really need to do here is to disable posting of receives on this endpoint.

Reply:
Thank you! Can you check the latest version